sqlite_query.c

```diff
@@ -412,6 +413,98 @@ sqlite_convert_to_pg(Form_pg_attribute att, sqlite3_value * val, AttInMetadata *
 				}
 				break;
 			}
+		case MACADDROID:
+			{
+				switch (sqlite_value_affinity)
+				{
+					/* SQLite MAC address output always normalized to int64.
+					 * In sqlite_data_norm.c there is special additional C function.
+					 */
+					case SQLITE_INTEGER: /* <-- proper and recommended SQLite affinity of value for pgtyp */
+						{
+							sqlite3_int64		 value = sqlite3_value_int64(val);
+							const sqlite3_uint64 max = (1ULL << (6 * CHAR_BIT)) - 1ULL; /* maximal int64 for macaddr 6b*/
+							macaddr*			 retval;
+							if (value > max )
+								ereport(ERROR,
+										 (errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+										  errmsg("int64 for macaddr out of range")));
+							retval = (macaddr *) palloc(sizeof(macaddr));
+							retval->a = (value >> (CHAR_BIT * 5)) & 0xFF;
+							retval->b = (value >> (CHAR_BIT * 4)) & 0xFF;
+							retval->c = (value >> (CHAR_BIT * 3)) & 0xFF;
+							retval->d = (value >> (CHAR_BIT * 2)) & 0xFF;
+							retval->e = (value >> (CHAR_BIT * 1)) & 0xFF;
+							retval->f = (value >> (CHAR_BIT * 0)) & 0xFF;
+							return (struct NullableDatum){MacaddrPGetDatum(retval), false};
+							break;
+						}
+					case SQLITE_FLOAT:
+					case SQLITE_BLOB:
+						{
+							sqlite_value_to_pg_error();
+							break;
+						}
+					case SQLITE3_TEXT:
+						{
+							if (value_byte_size_blob_or_utf8)
+								sqlite_value_to_pg_error();
+							else
+								pg_column_void_text_error();
+							break;
+						}
+					default:
+						{
+							sqlite_value_to_pg_error();
+							break;
+						}
+				}
+				break;
+			}
+		case MACADDR8OID:
+			{
+				switch (sqlite_value_affinity)
+				{
+					/* SQLite MAC address output always normalized to int64.
+					 * In sqlite_data_norm.c there is special additional C function.
+					 */
+					case SQLITE_INTEGER: /* <-- proper and recommended SQLite affinity of value for pgtyp */
+						{
+							sqlite3_int64	value = sqlite3_value_int64(val);
+							macaddr8 *		retval = (macaddr8 *) palloc(sizeof(macaddr8));
+							retval->a = (value >> (CHAR_BIT * 7)) & 0xFF;
+							retval->b = (value >> (CHAR_BIT * 6)) & 0xFF;
+							retval->c = (value >> (CHAR_BIT * 5)) & 0xFF;
+							retval->d = (value >> (CHAR_BIT * 4)) & 0xFF;
+							retval->e = (value >> (CHAR_BIT * 3)) & 0xFF;
+							retval->f = (value >> (CHAR_BIT * 2)) & 0xFF;
+							retval->g = (value >> (CHAR_BIT * 1)) & 0xFF;
+							retval->h = (value >> (CHAR_BIT * 0)) & 0xFF;
+							return (struct NullableDatum){Macaddr8PGetDatum(retval), false};
+							break;
```

Comment:
This line will never reach.

Reply:
This is something from `gcc` warning suppress C code infrastructure. This line like other similar in my PR about PostGIS.

Reply:
Could you show me the warning? Which version of gcc do you use?  
It's strange, because return will end the function immediately, so break will never reach.

Reply:
```
gcc --version
gcc (Debian 12.2.0-14) 12.2.0
```
Warning: `this code will never reach`
Testing context: `switch` with `return` without `break`.

Reply:
After code modification I can remove one of `break` without warnings, please check current implementation of the problem place, @son-phamngoc

Reply:
Thank you. Confirmed. Please help me to resolve this comment.